File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        int lIndex = 0, rIndex = nums.size() - 1, ret = nums[0];
        while (lIndex <= rIndex){
            if (nums[lIndex] < nums[rIndex]){
                ret = min(ret, nums[lIndex]);
                break;
            }

            int mid = (rIndex - lIndex) / 2 + lIndex;
            ret = min(ret, nums[mid]);

            if (nums[mid] >= nums[lIndex]){
                lIndex = mid + 1;
            }
            else{
                rIndex = mid - 1;
            }

        }

        return ret;
    }
};

File: Data Structures & Algorithms/maximum-product-subarray/submission-1.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        ret = max(nums)
        curMin, curMax = 1, 1

        for n in nums:
            tmp = n * curMax
            curMax = max(tmp, n * curMin, n)
            curMin = min(tmp, n * curMin, n)
            ret = max(ret, curMax)

        return ret